File: Leetcode/res/Longest Palindrome/2.cpp
\*
Author: allannozomu
Runtime: 4 ms
Memory: 8.9 MB*\

class Solution {
public:
    int longestPalindrome(string s) {
        unordered_map<char, int> count;
        
        int res = 0;
        
        for (char c : s){
            if (count[c]) {
                count[c] = 0;
                res += 2;
            } else {
                count[c] = 1;
            }
        }
        
        for (char c : s){
            if (count[c]) return res + 1;
        }
        return res;
    }
};

File: Leetcode/res/Longest Palindromic Substring/2.py
\*
Author: allannozomu
Runtime: 9120 ms
Memory: 109.1 MB*\

class Solution:
    dp = {}
    longest = (0,0)
    
    def longestPalindrome(self, s: str) -> str:
        dp = {}
        longest = (0,0)
        s = s
        for i in range(len(s) - 1):
            dp[i,i] = True
            if s[i] == s[i + 1]:
                longest = (i, i+1)
                dp[i,i+1] = True
            else:
                dp[i,i+1] = False
        dp[len(s) - 1, len(s) - 1] = True

        for i in range(2, len(s)):
            for j in range(len(s) - i):
                if (s[j] == s[j + i] and dp[j + 1, j + i - 1]):
                    dp[j, j+ i] = True
                    longest = (j, j+i)
                else:
                    dp[j, j+ i] = False
        return s[longest[0] : longest[1] + 1]